scripts/git_changes_utils.py

```diff
@@ -0,0 +1,304 @@
+# Copyright 2024 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Utils file for dealing with git changes."""
+
+from __future__ import annotations
+
+import collections
+import pprint
+import subprocess
+import sys
+
+from scripts import common
+
+from typing import Dict, List, Optional, Tuple
+
+
+GitRef = collections.namedtuple(
+    'GitRef', ['local_ref', 'local_sha1', 'remote_ref', 'remote_sha1'])
+FileDiff = collections.namedtuple('FileDiff', ['status', 'name'])
+
+
+def get_local_git_repository_remote_name() -> Optional[bytes]:
+    """Get the remote name of the local repository.
+
+    Returns:
+        Optional[bytes]. The remote name of the local repository.
+
+    Raises:
+        ValueError. Subprocess failed to start.
+        Exception. Upstream not set.
+    """
+    remote_name = b''
+    remote_num = 0
+    task = subprocess.Popen(
+        ['git', 'remote'], stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    out, err = task.communicate()
+    remotes = out[:-1].split(b'\n')
+    if not err:
+        for remote in remotes:
+            get_remotes_url_cmd = (
+                b'git config --get remote.%s.url' % remote).split()
+            task = subprocess.Popen(
+                get_remotes_url_cmd, stdout=subprocess.PIPE,
+                stderr=subprocess.PIPE)
+            remote_url, err = task.communicate()
+            if not err:
+                if remote_url.endswith(b'oppia/oppia.git\n'):
+                    remote_num += 1
+                    remote_name = remote
+            else:
+                raise ValueError(err)
+    else:
+        raise ValueError(err)
+
+    if not remote_num:
+        raise Exception(
+            'Error: Please set the git upstream.\n'
+            'To do that follow these steps:\n'
+            '1. Run the command \'git remote -v\'\n'
+            '2a. If upstream is listed in the command output, then run the '
```

Comment:
```suggestion
            '2a. If \'upstream\' is listed in the command output, then run the '
```

When you use "upstream" as a string that devs should look for, please put it in quotes to make your meaning clear. Ditto elsewhere

Reply:
Done.